src/Symfony/Component/Translation/Translator.php

```diff
@@ -419,8 +419,18 @@ private function getFallbackContent(MessageCatalogue $catalogue)
 
     private function getCatalogueCachePath($locale)
     {
+        $resources = array();
+        if (isset($this->resources[$locale])) {
+            foreach ($this->resources[$locale] as $resource) {
+                if (is_object($resource[1]) && !$resource[1] instanceof \Serializable) {
+                    $resource[1] = get_class($resource[1]);
```

Comment:
Not sure if this is relevant, but if we are looking for something unique here, it might be better to use `spl_object_hash($resource[1])`